### PR TITLE
feature(goUsers): Implements the goUsers feature

### PR DESCRIPTION
**Pull request by mattcreager**

### CODE REVIEW

**Comment by mattcreager** [comment]
@colinmacdonald Flawless Victory.

File: docs/v1/model/index.md [hunk 1]
```diff
@@ -24,30 +26,81 @@ introductory examples (based on the universal todo app) to help you get started.
 |:--|
 | Creates a new model instance, with a relative key. |
 
-| [$set(value)](./set.md)|
```
Comment (mattcreager) [inline docs/v1/model/index.md:27]: Let's put these methods in the same order as the menu: key/emit/sync/on/off

File: docs/v1/model/index.md [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 It encapsulates your application's data and provides an API to access, manipulate
 and persist that data.
 
-This page contains an index of the methods available on a model and some
-introductory examples (based on the universal todo app) to help you get started.
+This page contains an index of the three models created by GoAngular, methods
+available on the models, and some introductory examples (based on the universal
+todo app) to help you get started.
 
 ## Table of Contents
```
Comment (mattcreager) [inline docs/v1/model/index.md:12]: from what I can tell, the TOC doesn't work anymore!

File: docs/v1/model/index.md [hunk 3]
```diff
@@ -24,30 +26,81 @@ introductory examples (based on the universal todo app) to help you get started.
...
+|:--|
+| Adds an event handler. Can be used to monitor when a model is ready or errors. |
+
+| [$off(eventName, handler)](./off.md)|
+|:--|
+| Removes an event handler. |
+
+### [Key Model](./key_model/index.md)
+
+[$goKey](../key.md) returns a key model, which inherits all of the above methods.
```
Comment (mattcreager) [inline docs/v1/model/index.md]: mm.. I think we might want to be clear: inherits the above methods from the base [Model] & extends it with the following methods: or something.

File: docs/v1/model/index.md (hunk 2 above)
Comment (colinmacdonald) [inline docs/v1/model/index.md:12]: Looks like it never worked! Fixed.

File: docs/v1/model/index.md (hunk 1 above)
Comment (colinmacdonald) [inline docs/v1/model/index.md:27]: Done.

File: lib/connection.js [hunk 4]
```diff
@@ -136,7 +139,39 @@ Connection.prototype.$$connect = function() {
...
   this.$$connecting = true;
-  this.$$connection = this.$$conn.connect(user || {}, function() {});
+
+  var self = this;
+  this.$$connection = this.$$conn.connect(user || {}, function() {
+    self.isGuest = self.$$conn.isGuest();
+  });
+};
+
+Connection.prototype.$loginUrl = function(providers, returnTo) {
```
Comment (mattcreager) [inline lib/connection.js:149]: inline docs where you be?

**Comment by mattcreager** [comment]
LGTM!

File: tests/users_sync.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
+/* jshint browser:true */
+/* global require */
+
+describe('GoAngular.usersSync', function() {
+
+  'use strict';
+
+  var assert = require('gi-assert');
```
Comment (mattcreager) [inline tests/users_sync.js:8]: Could you create an issue to fix these up?

**Pull request merged**